tests/integration/helpers/no-op-test.js

```diff
@@ -0,0 +1,15 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Helper | no-op', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('It successfully renders and does nothing when clicked', async function (assert) {
+    assert.expect(0);
+
+    await render(hbs`<PolarisButton @onClick={{fn (no-op)}}></PolarisButton> `);
+    await click('button');
```

Comment:
Perhaps I'm missing something - how does this this test that the helper no-ops? 🤔

Reply:
welp, we're just testing that there's no assertion, not sure what more can be done to test this

Reply:
>not sure what more can be done to test this

Me neither 😝 But for there to be an assertion, that'd mean that `no-op` itself would have to contain an `assert`, which wouldn't make any sense anyway. Perhaps we can just make sure it doesn't throw any exception when clicked instead?

Reply:
> Perhaps we can just make sure it doesn't throw any exception when clicked instead?

yeah, exactly. I think that's already implied, since tests pass. If there would be an exception, tests would fail.

Reply:
>I think that's already implied, since tests pass. If there would be an exception, tests would fail.

Not quite though right? Wouldn't the tests _error_ instead of fail? Isn't it better to be explicit about what we're testing, instead of saying "there aren't any assertions in this test"?

Reply:
> Wouldn't the tests error instead of fail?

Not sure what you mean by this, but yes, an unexpected error would fail the tests. Example below with a reference error thrown in the helper
<img width="637" alt="Screen Shot 2020-11-17 at 10 34 11 AM" src="https://user-images.githubusercontent.com/160955/99365831-7794bb80-28c0-11eb-9a11-4a4f0101da14.png">

> Isn't it better to be explicit about what we're testing, instead of saying "there aren't any assertions in this test"?

Sorry, I 💯 agree. I just can't think of anything better. Please do suggest something if you happen to know.

Reply:
Hmm, that error message isn't helpful as it stands. Can we wrap the `click()` in a try-catch and assert if the click call blows up?

```js
try {
  await click('button');
} catch (err) {
  assert.notOk(true, 'Expected no-op not to throw when invoked');
}
```

Not a big deal if this doesn't work/make sense, just thought it'd be good to keep the tests as close to what we actually want to test as possible 🤷‍♂️

Reply:
> Hmm, that error message isn't helpful as it stands

Many errors are not helpful in JS, from my experience at least.

> Not a big deal if this doesn't work/make sense, just thought it'd be good to keep the tests as close to what we actually want to test as possible

I totally understand, no worries. I just fail to see how that's any better or closer to what we test